lib/slack-handler.js

```diff
@@ -26,17 +34,21 @@ var SLACK_EVENTS = {
 
     // Prepare IRC connection
     var user = context.slack.client.data.getUserById(context.slack.client.rtm.activeUserId);
+    context.logger.debug('IRC config -> '+user.profile.title);
     var userConfig = HELPERS.parseConfig(user.profile.title);
     // TODO: Fetch channel config from channel purpose
     var userChannels = _.chain(context.slack.client.data.channels)
                     .filter('is_member')
                     .map(HELPERS.getIrcChannelName.bind(null, context))
+                    .filter(function(channel) {
+                      return channel.name !== '';
+                    })
                     .valueOf();
 
     context.irc.options = {
       server: userConfig.server,
       port: userConfig.port || 6667,
-      nick: user.name,
+      nick: userConfig.userName || user.name,
```

Comment:
Not sure why there should be two ways of configuring the IRC nickname. Personally I find the JSON encoded data in the purpose fields really dirty. Unfortunately they seem necessary at the moment.